AbletonMCP_Remote_Script/handlers/clips.py

```
@@ -0,0 +1,800 @@
+    """
+    try:
+        clip = _get_clip(song, track_index, clip_index)
+        if not clip.is_audio_clip:
+            raise ValueError("Clip is not an audio clip")
+        conversion_type = str(conversion_type).lower()
+        if conversion_type not in ("drums", "harmony", "melody"):
+            raise ValueError("conversion_type must be 'drums', 'harmony', or 'melody'")
+        try:
+            from Live.Conversions import audio_to_midi_clip, AudioToMidiType
+        except ImportError:
+            raise Exception("Audio-to-MIDI conversion requires Live 12+")
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)